cpp/test/sg/rf_accuracy_test.cu

```diff
@@ -0,0 +1,149 @@
+    CUDA_CHECK(cudaStreamCreate(&stream));
+    handle.reset(new cumlHandle(1));
+    handle->setStream(stream);
+    auto allocator = handle->getDeviceAllocator();
+    setRFParams();
+    X_train = (T *)allocator->allocate(params.n_rows_train * sizeof(T), stream);
+    y_train =
+      (int *)allocator->allocate(params.n_rows_train * sizeof(int), stream);
+    X_test = (T *)allocator->allocate(params.n_rows_test * sizeof(T), stream);
+    y_test =
+      (int *)allocator->allocate(params.n_rows_test * sizeof(int), stream);
+    y_pred =
+      (int *)allocator->allocate(params.n_rows_test * sizeof(int), stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+  }
+
+  void TearDown() override {
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    auto allocator = handle->getDeviceAllocator();
+    allocator->deallocate(X_train, params.n_rows_train * sizeof(T), stream);
+    allocator->deallocate(y_train, params.n_rows_train * sizeof(int), stream);
+    allocator->deallocate(X_test, params.n_rows_test * sizeof(T), stream);
+    allocator->deallocate(y_test, params.n_rows_test * sizeof(int), stream);
+    allocator->deallocate(y_pred, params.n_rows_test * sizeof(int), stream);
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    CUDA_CHECK(cudaStreamDestroy(stream));
+    handle.reset();
+    rng.reset();
+  }
+
+  void runTest() {
+    for (int i = 0; i < params.n_reps; ++i) {
+      loadData(X_train, y_train, params.n_rows_train, 1);
+      loadData(X_test, y_test, params.n_rows_test, 1);
+      CUDA_CHECK(cudaStreamSynchronize(stream));
+      auto accuracy = runTrainAndTest();
+      ASSERT_GT(accuracy, params.min_expected_acc) << " @repetition=" << i;
+    }
+  }
+
+ private:
+  void setRFParams() {
+    DecisionTree::DecisionTreeParams tree_params;
+    auto algo = SPLIT_ALGO::GLOBAL_QUANTILE;
+    auto sc = CRITERION::CRITERION_END;
+    set_tree_params(tree_params, 1, /* max_depth */
+                    -1,             /* max_leaves */
+                    1.0,            /* max_features */
+                    16,             /* n_bins */
+                    algo,           /* split_algo */
+                    2,              /* min_rows_per_node */
+                    0.f,            /* min_impurity_decrease */
+                    false,          /* bootstrap_features */
+                    sc,             /* split_criterion */
+                    false,          /* quantile_per_tree */
+                    false           /* shuffle_features */
+    );
+    set_all_rf_params(rfp, 1, /* n_trees */
+                      true,   /* bootstrap */
+                      1.0,    /* rows_sample */
+                      -1,     /* seed */
+                      1,      /* n_streams */
+                      tree_params);
+  }
+
+  void loadData(T *X, int *y, int nrows, int ncols) {
+    rng->uniform(X, nrows * ncols, T(-1.0), T(1.0), stream);
+    rng->bernoulli<float, int>(y, nrows, params.pct_zero_class, stream);
```

Comment:
Ah, thank you for replacing the hacky version I had! Should this be (1.0 - pct_zero_class)? Or maybe rename the variable to pct_positive_class?

Reply:
`params.pct_zero_class` is correct, as the `bernoulli` method returns a value of 0 with this probability.

Reply:
thanks, makes sense!